src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/membership_table.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accounts
+
+import (
+	"database/sql"
+)
+
+const membershipSchema = `
+-- Stores data about users memberships to rooms.
+CREATE TABLE IF NOT EXISTS memberships (
+    -- The Matrix user ID localpart for the member
+    localpart TEXT NOT NULL,
+    -- The room this user is a member of
+    room_id TEXT NOT NULL,
+    -- The ID of the join membership event
+    event_id TEXT NOT NULL,
+
+    -- A user can only be member of a room once
+    PRIMARY KEY (localpart, room_id)
+);
+`
+
+const insertMembershipSQL = "" +
+	"INSERT INTO memberships(localpart, room_id, event_id) VALUES ($1, $2, $3)"
+
+const selectMembershipSQL = "" +
+	"SELECT * from memberships WHERE localpart = $1 AND room_id = $2"
+
+const selectMembershipByEventIDSQL = "" +
+	"SELECT localpart, room_id FROM memberships WHERE event_id = $1"
+
+const selectMembershipsByLocalpartSQL = "" +
+	"SELECT room_id FROM memberships WHERE localpart = $1"
+
+const deleteMembershipSQL = "" +
+	"DELETE FROM memberships WHERE localpart = $1 AND room_id = $2"
+
+const deleteMembershipByEventIDSQL = "" +
+	"DELETE FROM memberships WHERE event_id = $1"
```

Comment:
You can do multiple deletes in a single SQL statement on postgresql by using `ANY` and passing an array of strings.
There is an example of this in the federation sender https://github.com/matrix-org/dendrite/blob/master/src/github.com/matrix-org/dendrite/federationsender/storage/joined_hosts_table.go#L50